cmd/erasure-multipart.go

```diff
@@ -196,6 +196,24 @@ func (er erasureObjects) deleteAll(ctx context.Context, bucket, prefix string) {
 func (er erasureObjects) cleanupStaleUploadsOnDisk(ctx context.Context, disk StorageAPI, expiry time.Duration) {
 	drivePath := disk.Endpoint().Path
 
+	readDirFn(pathJoin(drivePath, minioMetaMultipartBucket), func(shaDir string, typ os.FileMode) error {
+		vi, err := disk.StatVol(ctx, pathJoin(minioMetaMultipartBucket, shaDir))
+		if err != nil {
+			return nil
+		}
+		if time.Since(vi.Created) < expiry {
+			return nil
+		}
+		dirs, err := disk.ListDir(ctx, "", minioMetaMultipartBucket, shaDir, 1)
+		if err != nil || len(dirs) > 0 {
+			return nil
+		}
+		wait := deleteMultipartCleanupSleeper.Timer(ctx)
+		er.deleteAll(ctx, minioMetaMultipartBucket, shaDir)
+		wait()
+		return nil
+	})
```

Comment:
this entire code that is added is literally being done the same manner below what is the difference in this code?

Reply:
Sorry, I tested based on an earlier version and found that there was a problem with empty directories not being deleted. The latest trunk has fixed this problem and I will close my pr.